Sierra/challenge_14/book_challenge.py

```diff
@@ -0,0 +1,120 @@
+'''
+print('PART 1\n')
+my_library = Booklist()
+print(my_library.books)
+print(type(my_library))
+
+'''
+Part #2:
+In book_list_class.py:
+Define the add() method to add a book to an object of class Booklist
+-The method should take 2 parameters other than self, both strings -- 'title', and 'author'
+-The method should create a Book object (it's already imported into the file) with those two attributes
+-Then, the method should append this book to books attribute of the the Booklist object  - i.e. self.books.append(book)
+
+In this script:
+Once you have finished the method, add the following books to your library:
+Just Mercy - Bryan Stevenson
+The New Jim Crow - Michelle Alexander
+The Truths We Hold - Kamala Harris
+My Grandmother's Hands - Resmaa Menakem
+
+Finally, print the books attribute of my_library to make sure your books have been added!
+'''
+print('\nPart 2\n')
+my_library.add('Just Mercy', 'Bryan Stevenson')
+my_library.add('The New Jim Crow', 'Michelle Alexander')
+my_library.add('The Truths We Hold', 'Kamala Harris')
+my_library.add('My Grandmother\'s Hands', 'Resmaa Menakem')
+
+print(my_library.books)
+
+'''
+Part #3:
+In book_list_class.py:
+Define the count_books() method to get the number of books in an object of class Booklist
+-the method only needs the self parameter
+-the method should return an integer that is the length of the list stored in the books attribute
+
+In this script:
+Once you have finished the method, count the books in my_library and print out the result
+'''
+print('\nPart 3\n')
+# I'm assuming the 'None' after the result is because we had to format the 'add()' method?!
+# Also, when I had my .append formatted with an f-string, the len(self.books) was correct with a count of 4
+# but now it says 8 because I had to change the formatting.
+print(my_library.count_books())
+
+'''
+Part #4:
+In book_list_class.py:
+Define the remove_title() method which will remove a book by its title from an object of class Booklist
+-the method should take the self parameter, plus an additional paramter 'title' (a string for the title of the book to remove)
+-the method should remove any books matching the input title from the Booklist
+-it does not need to return anything
+
+In this script:
+Once you have finished the method, remove 'Just Mercy' from my_library
+Then, print out the books attribute to make sure that book is gone
+'''
+print('\nPart 4\n')
+my_library.remove_title('Just Mercy') 
+print(my_library.books)
+
+
+'''
+Part #5:
+In book_list_class.py:
+Instantiate another object of class Booklist called nyt_bestsellers
+Then, add 2 books of your choice from the New York Times best sellers lists to nyt_bestsellers using the .add() method
+You can find NYT books here: https://www.nytimes.com/books/best-sellers/
+
+In this script:
+Then, print out the books attribute of nyt_bestsellers
+'''
+# This doesn't work here but it works in the book_list_class.py script!
+#print(nyt_bestsellers.books)
```

Comment:
For this part, you are supposed to instantiate the booklist objects first, and then add books to it, and then print them out. So basically, there are three things you have to write in this section.

Reply:
It says to do those steps in the book_list_class.py script so I thought I was supposed to do it in there /: that's where I did those steps